File: Creational/Lab2/Python/house.py
from real_estate import *

class House(RealEstate):
    def __str__(self):
        return f"House:\n{super().__str__()}"
    
    def clone(self):
        clone = copy.deepcopy(self)
        clone.realtor = Realtor(self.realtor.id, self.realtor.name_and_surname, self.realtor.phone, self.realtor.email)
        return clone

    def shallow_copy(self):
        return copy.copy(self)
        

File: Creational/Lab2/C#/House.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace C_
{
    public class House : RealEstate
    {
        public House(string address, string description, double area, double price, string status): 
            base(address, description, area, price, status) 
        { }

        public override string ToString()
        {
            return "House:\n" + base.ToString();
        }

        public override House Clone()
        {
            var clone = (House)MemberwiseClone();
            clone.Realtor = new Realtor(Realtor);
            return clone;
        }

        public override House ShallowCopy()
        {
            return (House)MemberwiseClone();
        }
    }
}
